source/postprocess/visualization.cc

```diff
@@ -97,6 +97,36 @@ namespace aspect
             return update_values;
           }
       };
+
+      /**
+       * This Postprocessor will generate the output variables of mesh velocity
+       * for when a free surface is used.
+       */
+      template <int dim>
+      class FreeSurfacePostprocessor: public DataPostprocessorVector< dim >, public SimulatorAccess<dim>
+      {
+        public:
+          FreeSurfacePostprocessor ()
+            : DataPostprocessorVector<dim>( "mesh_velocity", UpdateFlags(update_values) )
+          {}
+
+          virtual
+          void
+          compute_derived_quantities_vector (const std::vector<Vector<double> >              &uh,
+                                             const std::vector<std::vector<Tensor<1,dim> > > &,
+                                             const std::vector<std::vector<Tensor<2,dim> > > &,
+                                             const std::vector<Point<dim> > &,
+                                             const std::vector<Point<dim> > &,
+                                             std::vector<Vector<double> >                    &computed_quantities) const
+          {
+            const double velocity_scaling_factor =
+              this->convert_output_to_years() ? year_in_seconds : 1.0;
+            const unsigned int n_q_points = uh.size();
+            for (unsigned int q=0; q<n_q_points; ++q)
+              for (unsigned int i=0; i<computed_quantities[q].size(); ++i)
+                computed_quantities[q][i]=uh[q][i] * ((i < dim) ? velocity_scaling_factor : 1.0);
```

Comment:
`uh[q]` will have more than `dim` components, but `computed_quantities[q]` should have exactly `dim` components. At least that's what you say when you derive from `DataPostprocessorVector`.